Tasks/Gulp/gulptask.ts

```diff
@@ -2,118 +2,70 @@
 
 import path = require('path');
 import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
 
-tl.setResourcePath(path.join(__dirname, 'task.json'));
+async function executeTask() {
+	try {
+		tl.setResourcePath(path.join(__dirname, 'task.json'));
+		var gulpFilePath = tl.getDelimitedInput('gulpFile', '\n', true);
+		var gulp = tl.which('gulp', false);
+		var cwd = tl.getPathInput('cwd', true, false);
+		tl.mkdirP(cwd);
```

Comment:
if cwd is provided by user, then use the gulp option to take cwd as changing directory to it doesn't work